missed-moment.py

```diff
@@ -41,20 +37,189 @@ def capture_video():
         logging.error(f'Unhandled exception uploading files - {e}')
 
 
-def main():
-    logging.basicConfig(level=logging.INFO)
-    if not exists(MEDIA_DIR):
-        check_call(['sudo', 'mkdir', expanduser(MEDIA_DIR)])
-        # add write permissions for all
-        check_call(['sudo', 'chmod', 'a+w', expanduser(MEDIA_DIR)])
-    camera.start_recording(stream, format='h264')
-    logging.info('missed-moment ready to save a moment!')
-    button.when_pressed = capture_video
+def capture_audio(file_name):
+    # Write current stream to file
+    clean_file = f'{MEDIA_DIR}/{file_name}.wav'
+    logging.debug(f'audio_clean_file:{clean_file}')
+
+    # get size of the default time machine file, and as soon as it is bigger we know 
+    # time machine has started recording to file
+    result = Popen(f"ls -la {AUDIO_CAPTURE_TEMP_FILENAME} | awk '{{print $5}}'", shell=True, stdout=subprocess.PIPE)
+    default_file_size_string = result.stdout.read().decode('utf-8')
+    logging.debug(f'initial {AUDIO_CAPTURE_TEMP_FILENAME} size:{default_file_size_string}')
+
+    # timemachine start and jack_capture stop to get the audio ring buff stream saved to file
+    # don't think I can use subprocess.call to wait because this is a oscsend command
+    command = f"oscsend localhost {AUDIO_CAPTURE_REMOTE_PORT} /jack_capture/tm/start"
+    system(command)
+
+    file_saved = False
+    while not file_saved:
+        result = Popen(f"ls -la {AUDIO_CAPTURE_TEMP_FILENAME} | awk '{{print $5}}'", shell=True, stdout=subprocess.PIPE)
+        curr_file_size_string = result.stdout.read().decode('utf-8')
+        logging.debug(f'current {AUDIO_CAPTURE_TEMP_FILENAME} size:{curr_file_size_string}')
+        if (int(curr_file_size_string) > int(default_file_size_string)):
+            logging.debug(f'{AUDIO_CAPTURE_TEMP_FILENAME} saved')
+            file_saved = True
```

Comment:
So we assume the file save finished because the file size changed?
Not risk of getting that wrong?

Reply:
the audio capture tool (jack_capture) in ringbuffer mode, saves to memory, but must know what the filename it will save to.  The temp file seems to have an 80byte  filesize to it prior to the ringbuffer contents being saved to it.  to remotely control jack_capture, I have to send a "start" command to it, which I know has dumped the buffer to file as soon as the filesize is larger.  The file starts to grow (since the buffer is continually writing to it now) until I send a "stop" command to it.
To answer your question, yes we assume the file starts saving the file because the filesize changed.